jsonclient/backoff_test.go

```diff
@@ -0,0 +1,108 @@
+package jsonclient
+
+import (
+	"math"
+	"testing"
+	"time"
+)
+
+const testLeeway = 10 * time.Microsecond
+
+func fuzzyTimeEquals(a, b time.Time, leeway time.Duration) bool {
+	diff := math.Abs(float64(a.Sub(b).Nanoseconds()))
+	if diff < float64(leeway.Nanoseconds()) {
+		return true
+	}
+	return false
+}
+
+func fuzzyDurationEquals(a, b time.Duration, leeway time.Duration) bool {
+	diff := math.Abs(float64(a.Nanoseconds() - b.Nanoseconds()))
+	if diff < float64(leeway.Nanoseconds()) {
+		return true
+	}
+	return false
+}
+
+func TestBackoff(t *testing.T) {
+	b := backoff{}
+
+	// Test that the interval increases as expected
+	for i := uint(0); i < maxMultiplier; i++ {
+		n := time.Now()
+		interval := b.set(nil)
+		if interval != time.Second*(1<<i) {
+			t.Fatalf("backoff.set() returned an unexpected duration. wanted: %s, got: %s", time.Second*(1<<i), interval)
+		}
+		expected := n.Add(interval)
+		if !fuzzyTimeEquals(expected, b.notBefore, testLeeway) {
+			t.Fatalf("backoff.notBefore is not expected time. wanted: %s (+/- 10µs), got: %s", expected, b.notBefore)
```

Comment:
This [looks to be failing](https://travis-ci.org/google/certificate-transparency-go/jobs/285813715) under `--race`...